lib/regular_expression/lexer.rb

```diff
@@ -4,7 +4,6 @@ module RegularExpression
   class Lexer
     SINGLE = {
       "^" => :CARET,
-      "$" => :ENDING,
```

Comment:
This is never returned from the parser because it's matched as an `:ANCHOR`. Also not used in the grammar, looks fine to remove.